ssl/ssl_lib.c

```diff
@@ -2506,6 +2510,57 @@ int ssl_write_internal(SSL *s, const void *buf, size_t num, size_t *written)
     }
 }
 
+int ssl_writev_internal(SSL *s, const OSSL_IOVEC *iov, size_t iovcnt,
+                        size_t *written)
+{
+    SSL_CONNECTION *sc = SSL_CONNECTION_FROM_SSL(s);
+
+#ifndef OPENSSL_NO_QUIC
+    if (IS_QUIC(s))
+        return s->method->ssl_write(s, buf, num, written);
+#endif
+
+    if (sc == NULL)
+        return 0;
+
+    if (sc->handshake_func == NULL) {
+        ERR_raise(ERR_LIB_SSL, SSL_R_UNINITIALIZED);
+        return -1;
+    }
+
+    if (sc->shutdown & SSL_SENT_SHUTDOWN) {
+        sc->rwstate = SSL_NOTHING;
+        ERR_raise(ERR_LIB_SSL, SSL_R_PROTOCOL_IS_SHUTDOWN);
+        return -1;
+    }
+
+    if (sc->early_data_state == SSL_EARLY_DATA_CONNECT_RETRY
+                || sc->early_data_state == SSL_EARLY_DATA_ACCEPT_RETRY
+                || sc->early_data_state == SSL_EARLY_DATA_READ_RETRY) {
+        ERR_raise(ERR_LIB_SSL, ERR_R_SHOULD_NOT_HAVE_BEEN_CALLED);
+        return 0;
+    }
+    /* If we are a client and haven't sent the Finished we better do that */
+    ossl_statem_check_finish_init(sc, 1);
+
+    if ((sc->mode & SSL_MODE_ASYNC) && ASYNC_get_current_job() == NULL) {
+        int ret;
+        struct ssl_async_args args;
+
+        args.s = s;
+        args.buf = (void *)iov;
+        args.num = iovcnt;
+        args.type = WRITEVFUNC;
+        args.f.func_writev = s->method->ssl_writev;
+
+        ret = ssl_start_async_job(s, &args, ssl_io_intern);
+        *written = sc->asyncrw;
+        return ret;
+    } else {
```

Comment:
Drop the else clause since it's returning.

Reply:
I also copy this function from `ssl_write_internal` where there is an else clause after return.

Reply:
As above.